src/Acuminator/Acuminator.Analyzers/StaticAnalysis/DacPropertyAttributes/MultipleAttributesOnDacPropertyFixBase.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Acuminator.Utilities.Common;
+using Acuminator.Utilities.Roslyn;
+using Acuminator.Utilities.Roslyn.PXFieldAttributes;
+using Acuminator.Utilities.Roslyn.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Acuminator.Analyzers.StaticAnalysis.DacPropertyAttributes
+{
+	/// <summary>
+	/// A multiple attributes on DAC property fix base class.
+	/// </summary>
+	public abstract class MultipleAttributesOnDacPropertyFixBase : CodeFixProvider
+	{
+		public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			string codeActionName = GetCodeActionName();
+			Func<FieldTypeAttributeInfo, bool> removePredicate = GetRemoveAttributeByAttributeInfoPredicate();
+
+			if (codeActionName.IsNullOrWhiteSpace() || removePredicate == null)
+				return;
+
+			SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken)
+													.ConfigureAwait(false);
+
+			if (!(root?.FindNode(context.Span) is AttributeSyntax attributeNode) || context.CancellationToken.IsCancellationRequested)
+				return;
+
+			var propertyDeclaration = attributeNode.Parent<PropertyDeclarationSyntax>();
+
+			if (propertyDeclaration == null)
+				return;
+
+			SemanticModel semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken)
+																.ConfigureAwait(false);
+			if (semanticModel == null)
+				return;
+
+			CodeAction codeAction =
+				CodeAction.Create(codeActionName,
+								  cToken => RemoveAllOtherAttributesFromPropertyAsync(context.Document, root, attributeNode,
+																					  propertyDeclaration, semanticModel, 
+																					  removePredicate, cToken),
+								  equivalenceKey: codeActionName);
+
+			context.RegisterCodeFix(codeAction, context.Diagnostics);
+		}
+
+		protected abstract string GetCodeActionName();
+
+		protected abstract Func<FieldTypeAttributeInfo, bool> GetRemoveAttributeByAttributeInfoPredicate();
+
+		private Task<Document> RemoveAllOtherAttributesFromPropertyAsync(Document document, SyntaxNode root, AttributeSyntax attributeNode,
+																		 PropertyDeclarationSyntax propertyDeclaration, SemanticModel semanticModel,
+																		 Func<FieldTypeAttributeInfo, bool> removePredicate, CancellationToken cancellationToken)
+		{
+			return Task.Run(() =>
```

Comment:
Using `Task.Run` for executing synchronous code should be avoided, according to VS manuals. It doesn't improve overall performance and can lead to thread pool starvation.

Reply:
@dropsonic the code action requires method which returns `Task<Document>`. Will it execute it asynchronously if I just use `Task.FromResult` to return `Task<Document>`?

Reply:
@SENya1990 yes, because the host (Visual Studio) runs code analysis as a background operation. There is no need to run new tasks explicitly - it doesn't provide any additional benefits.

Reply:
ok, will fix